Gems/Atom/Feature/Common/Code/Source/PostProcessing/WhiteBalancePass.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <PostProcessing/WhiteBalancePass.h>
+#include <PostProcess/PostProcessFeatureProcessor.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/Scene.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        RPI::Ptr<WhiteBalancePass> WhiteBalancePass::Create(const RPI::PassDescriptor& descriptor)
+        {
+            RPI::Ptr<WhiteBalancePass> pass = aznew WhiteBalancePass(descriptor);
+            return AZStd::move(pass);
+        }
+
+        WhiteBalancePass::WhiteBalancePass(const RPI::PassDescriptor& descriptor)
+            : RPI::ComputePass(descriptor)
+        {
+        }
+
+        bool WhiteBalancePass::IsEnabled() const
+        {
+            if (!ComputePass::IsEnabled())
+            {
+                return false;
+            }
+            const RPI::Scene* scene = GetScene();
+            if (!scene)
+            {
+                return false;
+            }
+            PostProcessFeatureProcessor* fp = scene->GetFeatureProcessor<PostProcessFeatureProcessor>();
+            const RPI::ViewPtr view = GetRenderPipeline()->GetDefaultView();
+            if (!fp)
+            {
+                return false;
+            }
+            PostProcessSettings* postProcessSettings = fp->GetLevelSettingsFromView(view);
+            if (!postProcessSettings)
+            {
+                return false;
+            }
+            const WhiteBalanceSettings* whiteBalanceSettings = postProcessSettings->GetWhiteBalanceSettings();
+            if (!whiteBalanceSettings)
+            {
+                return false;
+            }
+            return whiteBalanceSettings->GetEnabled();
+        }
+
+        void WhiteBalancePass::FrameBeginInternal(FramePrepareParams params)
+        {
+            // Must match the struct in WhiteBalance.azsl
+            struct Constants
+            {
+                AZStd::array<u32, 2> m_outputSize;
+                AZStd::array<float, 2> m_outputCenter;
+                float m_temperature = WhiteBalance::DefaultTemperature;
+                float m_tint = WhiteBalance::DefaultTint;
+            } constants{};
```

Comment:
Add a AZStd::array<float, 2>m_pad at the end to ensure that the struct ends at 16 byte boundary